### PR TITLE
Fix/remove TODO-NULLABLE in System.Private.Xml

### LINKED ISSUE #44144: XmlReader.BaseURI should be not nullable
Currently XmlReader.BaseURI in some exceptional situations (as found when applying nullable annotations) can return null, this seem to be due to bug in the code. BaseURI is documented as:

```
If there is no base URI for the nodes being returned (for example, they were parsed from an in-memory string), String.Empty is returned.
```

we should fix all corresponding TODO-NULLABLE when doing this work.

### PR DESCRIPTION
**Pull request by krwq**
Fixes: https://github.com/dotnet/runtime/issues/44144

- changes nullability of XmlReader.BaseUri to be non-nullable (with one code fix where it can return null and will now return string.Empty)
- removed TODO-NULLABLEs related to fixed bugs in roslyn or unactionable

### CODE REVIEW

File: src/libraries/System.Private.Xml/src/System/Xml/Dom/DocumentXPathNavigator.cs [hunk 1]
```diff
@@ -180,10 +180,6 @@ public override string Value
                         return ValueText;
                     default:
                         Debug.Assert(_source.Value != null);
-                        // TODO-NULLABLE: Consider switching this.Value to nullable even if that implies switching it in the base type.
```
Comment (krwq) [inline src/libraries/System.Private.Xml/src/System/Xml/Dom/DocumentXPathNavigator.cs:183]: this didn't seem possible to me in legitimate scenario and I found this not worth investigating without actual repro

File: src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReader.cs [hunk 2]
```diff
@@ -140,7 +140,7 @@ public virtual bool HasValue
         public abstract int Depth { get; }
 
         // Gets the base URI of the current node.
-        public abstract string? BaseURI { get; }
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReader.cs:143]: Doesn't the ref need to be updated as well?

Comment (stephentoub) [reply src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReader.cs:143]: (Note as well this needs to be added to the breaking change list for nullable annotations.)

**Review by stephentoub — APPROVED** [review]
Other than the ref, LGTM.  Note I made a few of these same changes in https://github.com/dotnet/runtime/pull/44136.

**Comment by krwq** [comment]
@stephentoub didn't notice your PR, fortunatelly just a very small overlap, will take a look at your PR

**Comment by krwq** [comment]
ok, merging - ony diff is ref/derived classes in other libs missing same annotation

**Comment by stephentoub** [comment]
> ony diff is ref/derived classes in other libs missing same annotation

Example?

**Comment by krwq** [comment]
Not sure what do you mean
https://github.com/dotnet/runtime/pull/44149/commits/4ae7913a1c6c237891829eb7696fb7009898dda3

- DataTextReader.BaseURI (System.Data.Common impl)
- XmlDictionaryReader.BaseURI (DataContractSerialization impl + ref)
- XmlValidatingReader.BaseURI (ref)
- XmlTextReader.BaseURI (ref)

**Comment by stephentoub** [comment]
Oh by diff you meant you were describing last commit.

**Comment by krwq** [comment]
yes

**Pull request merged**